src/algebra/monoid_algebra/basic.lean

```diff
@@ -548,14 +548,24 @@ lemma single_one_comm [comm_semiring k] [mul_one_class G] (r : k) (f : monoid_al
 by { ext, rw [single_one_mul_apply, mul_single_one_apply, mul_comm] }
 
 /-- `finsupp.single 1` as a `ring_hom` -/
-@[simps] def single_one_ring_hom [semiring k] [monoid G] : k →+* monoid_algebra k G :=
+@[simps] def single_one_ring_hom [semiring k] [mul_one_class G] : k →+* monoid_algebra k G :=
 { map_one' := rfl,
   map_mul' := λ x y, by rw [single_add_hom, single_mul_single, one_mul],
   ..finsupp.single_add_hom 1}
 
+/--  A multiplicative homomorphism `f : G →* H` between two monoids induces a ring
+homomorphism `map_domain_ring_hom k f : monoid_algebra k G →+* monoid_algebra k H`. -/
```

Comment:
To avoid repeating the type in the statement, perhaps
```suggestion
/-- If `f : G → H` is a multiplicative homomorphism between two monoids, then
`finsupp.map_domain f` is a ring homomorphism between their algebras. -/
```
(and similarly for the other docstrings)

This has the advantage of creating a link to `finsupp.map_domain`.

Reply:
I did this, except that I also changed `is a ring homomorphism between their **monoid** algebras`